### PR TITLE
Update to 1.12

### PR DESCRIPTION
**Pull request by krwminer**
Most of the low hanging fruit. 

I was interested in getting back into modding again so I took a look at bringing your mod to 1.12.
These are mostly name changes and such with one or two more moderate changes, (registry).
There are only 6 more files to fix before it will compile.

I wanted to offer it to you to save you time when you get around to porting to 1.12.

Feel free to integrate it, comment back to me or reject it. 

Thanks for making a great mod!

-krwminer

### CODE REVIEW

**Comment by BobMowzie** [comment]
Wow, this is so great. Thanks! I'm taking a small break right now but I'll look through it soon!

**Comment by krwminer** [comment]
Enjoy your break! We all need one from time to time!

I want to better understand the new way Forge wants us to register stuff.

If I have anything more I will post another pull request. Same thing, no pressure to accept, I am not trying to rush you... I am just having fun and want to share. :)

**Comment by Sunconure11** [comment]
I'm working on my own port. Anything you can gleam from my port, you can use in your port. Just beware that I changed the mappings.

**Comment by BobMowzie** [comment]
I am impressed with and grateful for both of your work. I’m afraid my hiatus stretches longer than I had intended it to while other projects take priority.

File: src/main/java/com/bobmowzie/mowziesmobs/server/sound/MMSounds.java [hunk 1]
```diff
@@ -2,108 +2,69 @@
...
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
 import net.minecraftforge.fml.common.registry.GameRegistry;
 
 import com.bobmowzie.mowziesmobs.MowziesMobs;
 
-public enum MMSounds {
-    INSTANCE;
-
+@Mod.EventBusSubscriber
```
Comment (pau101) [inline src/main/java/com/bobmowzie/mowziesmobs/server/sound/MMSounds.java]: All EventBusSubscribers not on the @Mod class need modid specified,@Mod.EventBusSubscriber(modid = MowziesMobs.MODID)

Comment (pau101) [inline src/main/java/com/bobmowzie/mowziesmobs/server/sound/MMSounds.java]: All EventBusSubscribers not on a @​Mod class need their modid specified:
```java
@Mod.EventBusSubscriber(modid = MowziesMobs.MODID)
```

File: src/main/java/com/bobmowzie/mowziesmobs/server/sound/MMSounds.java [hunk 2]
```diff
@@ -167,139 +109,209 @@
...
             EFFECT_GEOMANCY_RUMBLE_2,
             EFFECT_GEOMANCY_RUMBLE_3
     };
+    public static SoundEvent EFFECT_GEOMANCY_HIT_SMALL;
+    public static SoundEvent EFFECT_GEOMANCY_BOULDER_CHARGE;
+    public static SoundEvent EFFECT_GEOMANCY_MAGIC_CHARGE_SMALL;
+
+    @SubscribeEvent
+    public static void registerSoundEvents(RegistryEvent.Register<SoundEvent> event) {  	
+        LASER = registerSound(event, "laser");
```
Comment (pau101) [inline src/main/java/com/bobmowzie/mowziesmobs/server/sound/MMSounds.java]: This registry event usage is incorrect, use [Fairy Lights](https://github.com/pau101/Fairy-Lights/blob/ebea4689ea2707715111b4af67354ca8d3999667/src/main/java/com/pau101/fairylights/server/sound/FLSounds.java) as a reference for what the structure should be.

**Pull request merged**